### PR TITLE
CFE-3292: Removed soft fail related to the use of globs

### PR DESCRIPTION
**Pull request by larsewi**
Merge together:
* https://github.com/cfengine/enterprise/pull/786

TODO:
- [x] Manually test on Windows
- [x] Fix `findfiles_up.cf` acceptance test on windows

### CODE REVIEW

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10242/)

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10249/)

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10250/)

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10251/)

**Comment by cf-bottom** [comment]
Thank you for submitting a PR! Maybe @craigcomstock can review this?

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10272/)

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10275/)

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10300/)

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10301/)

**Comment by olehermanse** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10386/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10386/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10386/

**Comment by larsewi** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10396/configure)

**Comment by craigcomstock** [comment]
[Build Status](https://ci.cfengine.com/job/pr-pipeline/10411/)

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10412/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10412/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10412/

**Comment by craigcomstock** [comment]
hpux failure
```
19:11:04 ERROR: glob_lib_test.c:394 Failure!
```

suse-15 errors are all jenkins upgrade related, lack of 'sys' group and such.

**Comment by larsewi** [comment]
@cf-bottom, I tried to fix a bug on Windows, can I have a new build in Jenkins to test if it works?

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10417/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10417/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10417/

File: tests/acceptance/01_vars/02_functions/findfiles.cf [hunk 1]
```diff
@@ -76,27 +72,21 @@ bundle agent test
...
       "expected[e]" string => "";
 
-    !windows::
-      "expected[f]" string => "$(G.testdir)/tu/*";
-      "expected[g]" string => "$(G.testdir)/a,$(G.testdir)/bc,$(G.testdir)/d,$(G.testdir)/g,$(G.testdir)/klm,$(G.testdir)/tu,$(G.testdir)/d/e,$(G.testdir)/g/h,$(G.testdir)/klm/nop,$(G.testdir)/tu/*,$(G.testdir)/d/e/f,$(G.testdir)/g/h/i,$(G.testdir)/klm/nop/qrs,$(G.testdir)/g/h/i/j";
-      "expected[h]" string => "$(G.testdir)/g/h/i/j";
+      "expected[f]" string => "$(G.testdir)$(const.dirsep)tu$(const.dirsep)*";
+      "expected[g]" string => "$(G.testdir)$(const.dirsep)a,$(G.testdir)$(const.dirsep)bc,$(G.testdir)$(const.dirsep)d,$(G.testdir)$(const.dirsep)g,$(G.testdir)$(const.dirsep)klm,$(G.testdir)$(const.dirsep)tu,$(G.testdir)$(const.dirsep)d$(const.dirsep)e,$(G.testdir)$(const.dirsep)g$(const.dirsep)h,$(G.testdir)$(const.dirsep)klm$(const.dirsep)nop,$(G.testdir)$(const.dirsep)tu$(const.dirsep)*,$(G.testdir)$(const.dirsep)d$(const.dirsep)e$(const.dirsep)f,$(G.testdir)$(const.dirsep)g$(const.dirsep)h$(const.dirsep)i,$(G.testdir)$(const.dirsep)klm$(const.dirsep)nop$(const.dirsep)qrs,$(G.testdir)$(const.dirsep)g$(const.dirsep)h$(const.dirsep)i$(const.dirsep)j";
```
Comment (craigcomstock) [inline tests/acceptance/01_vars/02_functions/findfiles.cf:83]: I gotta wonder if `$(const.dirsep)` isn't the best idea... this policy is rather unreadable.

Maybe `/` being the standard and on windows we replace with `$(const.dirsep)` might be a better idea. Or even `\` replaced with `/` on unix so go either way. :p

**Review by craigcomstock — DISMISSED** [review]
Cool.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10490/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10490/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10490/

**Review by craigcomstock — APPROVED** [review]

**Comment by craigcomstock** [comment]
@cf-bottom jenkins please with exotics

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10493/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10493/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10493/

**Comment by larsewi** [comment]
@cf-bottom Jenkins with exotics please :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10498/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10498/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10498/

**Comment by larsewi** [comment]
HP-UX failure looks like rsync failures. Sequential test is marked as unstable.

**Comment by larsewi** [comment]
@cf-bottom Jenkins with exotics please :)

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10501/)

(with exotics)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10501/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10501/

**Comment by larsewi** [comment]
HP-UX is flaky rsync errors, otherwise all green 💚

**Review by olehermanse — APPROVED** [review]

**Comment by craigcomstock** [comment]
cherry picked to 3.21.x
https://github.com/cfengine/core/pull/5555

**Pull request merged**